### PR TITLE
mail-client/alot: Add version bump 0.9.1

### PR DESCRIPTION
**Pull request by GuillaumeSeren**

### CODE REVIEW

File: dev-python/urwidtrees/urwidtrees-1.0.3.ebuild [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
```
Comment (DarthGandalf) [inline dev-python/urwidtrees/urwidtrees-1.0.3.ebuild:4]: add an empty line below

File: dev-python/urwidtrees/urwidtrees-1.0.3.ebuild [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+PYTHON_COMPAT=( python{3_6,3_7,3_8} )
```
Comment (DarthGandalf) [inline dev-python/urwidtrees/urwidtrees-1.0.3.ebuild:5]: you could put `3_` outside of `{}`

File: mail-client/alot/alot-0.9.1.ebuild [hunk 3]
```diff
@@ -18,17 +19,16 @@ IUSE="doc test"
 RESTRICT="!test? ( test )"
 
 RDEPEND="
-	>=app-crypt/gpgme-1.9.0[python,${PYTHON_USEDEP}]
-	>=dev-python/configobj-4.7.0[${PYTHON_USEDEP}]
+	app-crypt/gpgme[python,${PYTHON_USEDEP}]
```
Comment (DarthGandalf) [inline mail-client/alot/alot-0.9.1.ebuild:22]: why did you remove `>=`?

File: dev-python/urwidtrees/urwidtrees-1.0.3.ebuild [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+inherit distutils-r1
+
+DESCRIPTION="Tree widgets for urwid"
+HOMEPAGE="https://github.com/pazz/urwidtrees"
+SRC_URI="https://github.com/pazz/urwidtrees/archive/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="GPL-3"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="doc"
```
Comment (janko-j1) [inline dev-python/urwidtrees/urwidtrees-1.0.3.ebuild:16]: You should probably use distutils_enable_sphinx, which appends the doc keyword to IUSE for you: you can drop this line.

File: dev-python/urwidtrees/urwidtrees-1.0.3.ebuild [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+	use doc && emake -C docs html
+}
+
+src_install() {
+	distutils-r1_src_install
+
+	if use doc; then
+		dodoc -r docs/build/html/.
+	fi
+}
```
Comment (janko-j1) [inline dev-python/urwidtrees/urwidtrees-1.0.3.ebuild:45]: Those lines can be replaced by a single distutils_enable_sphinx call

File: mail-client/alot/alot-0.9.1.ebuild [hunk 6]
```diff
@@ -18,17 +19,16 @@ IUSE="doc test"
...
-	>=dev-python/urwid-1.3.0[${PYTHON_USEDEP}]
-	>=dev-python/urwidtrees-1.0[${PYTHON_USEDEP}]
+	dev-python/urwid[${PYTHON_USEDEP}]
+	dev-python/urwidtrees[${PYTHON_USEDEP}]
 	>=dev-python/twisted-18.4[${PYTHON_USEDEP}]
 	net-mail/mailbase
-	>=net-mail/notmuch-0.27[crypt,python]
+	net-mail/notmuch[crypt,python]
 	"
 DEPEND="
```
Comment (janko-j1) [inline mail-client/alot/alot-0.9.1.ebuild:31]: This should be BDEPEND. It can be called after distutils_enable_tests, which automatically appends test? ( ${RDEPEND} ) to it.

File: mail-client/alot/alot-0.9.1.ebuild [hunk 7]
```diff
@@ -18,17 +19,16 @@ IUSE="doc test"
...
-	dev-python/setuptools[${PYTHON_USEDEP}]
 	doc? ( dev-python/sphinx[${PYTHON_USEDEP}] )
```
Comment (janko-j1) [inline mail-client/alot/alot-0.9.1.ebuild:32]: Drop this and use distutils_enable_sphinx

File: mail-client/alot/alot-0.9.1.ebuild [hunk 8]
```diff
@@ -18,17 +19,16 @@ IUSE="doc test"
...
 	test? (
```
Comment (janko-j1) [inline mail-client/alot/alot-0.9.1.ebuild:33]: append mock to BDEPEND after calling distutils_enable_tests

File: mail-client/alot/alot-0.9.1.ebuild [hunk 9]
```diff
@@ -37,22 +37,19 @@ DEPEND="
...
+	emake -C docs man
+	use doc && emake -C docs html
 }
 
 python_install_all() {
+	use doc && local HTML_DOCS=( docs/build/html/. )
+	doman docs/build/man/*
 	distutils-r1_python_install_all
 
 	insinto /usr/share/alot
```
Comment (janko-j1) [inline mail-client/alot/alot-0.9.1.ebuild:55]: These lines can be replaced by a call to distutils_enable_sphinx

File: mail-client/alot/alot-0.9.1.ebuild [hunk 10]
```diff
@@ -3,7 +3,8 @@
 
 EAPI=7
 
-PYTHON_COMPAT=( python3_{6,7} )
+DISTUTILS_USE_SETUPTOOLS=rdepend
+PYTHON_COMPAT=( python3_{6,7,8} )
```
Comment (janko-j1) [inline mail-client/alot/alot-0.9.1.ebuild:7]: Tests fail with py3.8 because notmuch does not have py3.8 yet. This is why the RESTRICT below?

File: dev-python/urwidtrees/urwidtrees-1.0.3.ebuild (hunk 1 above)
Comment (GuillaumeSeren) [reply dev-python/urwidtrees/urwidtrees-1.0.3.ebuild:4]: ok

File: dev-python/urwidtrees/urwidtrees-1.0.3.ebuild (hunk 2 above)
Comment (GuillaumeSeren) [reply dev-python/urwidtrees/urwidtrees-1.0.3.ebuild:5]: ok

File: mail-client/alot/alot-0.9.1.ebuild (hunk 3 above)
Comment (GuillaumeSeren) [reply mail-client/alot/alot-0.9.1.ebuild:22]: Just because the minimal available version in portage is sufficient (1.13.0).

File: dev-python/urwidtrees/urwidtrees-1.0.3.ebuild (hunk 4 above)
Comment (GuillaumeSeren) [reply dev-python/urwidtrees/urwidtrees-1.0.3.ebuild:16]: ok

File: dev-python/urwidtrees/urwidtrees-1.0.3.ebuild (hunk 5 above)
Comment (GuillaumeSeren) [reply dev-python/urwidtrees/urwidtrees-1.0.3.ebuild:45]: ok

File: mail-client/alot/alot-0.9.1.ebuild (hunk 10 above)
Comment (janko-j1) [reply mail-client/alot/alot-0.9.1.ebuild:7]: Nevermind: https://github.com/gentoo/gentoo/pull/16668 will add py3.8 to notmuch

File: mail-client/alot/alot-0.9.1.ebuild (hunk 6 above)
Comment (GuillaumeSeren) [reply mail-client/alot/alot-0.9.1.ebuild:31]: Ok.

File: mail-client/alot/alot-0.9.1.ebuild (hunk 10 above)
Comment (GuillaumeSeren) [reply mail-client/alot/alot-0.9.1.ebuild:7]: Cool

File: mail-client/alot/alot-0.9.1.ebuild (hunk 7 above)
Comment (GuillaumeSeren) [reply mail-client/alot/alot-0.9.1.ebuild:32]: As discussed on irc this helper does not work when building multiple (here man page),
sphinx target.

File: mail-client/alot/alot-0.9.1.ebuild (hunk 8 above)
Comment (GuillaumeSeren) [reply mail-client/alot/alot-0.9.1.ebuild:33]: I did not get it at first now it is ok, thank you.

File: mail-client/alot/alot-0.9.1.ebuild (hunk 9 above)
Comment (GuillaumeSeren) [reply mail-client/alot/alot-0.9.1.ebuild:55]: Unfortunately no.

File: mail-client/alot/alot-0.9.1.ebuild (hunk 7 above)
Comment (janko-j1) [reply mail-client/alot/alot-0.9.1.ebuild:31]: I think you should put this as a comment in the ebuild so that one knows why distutils_enable_sphinx is not used

File: mail-client/alot/alot-0.9.1.ebuild [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+	"
+
+PATCHES=(
+	"${FILESDIR}/0.9-0001-remove-non-working-test.patch"
+	)
+
+distutils_enable_tests unittest
+
+BDEPEND+=" test? ( dev-python/mock[${PYTHON_USEDEP}] ) "
+
```
Comment (juippis) [inline mail-client/alot/alot-0.9.1.ebuild:41]:
```
tests.widgets.test_globals (unittest.loader._FailedTest) ... ERROR

======================================================================
ERROR: alot.buffers (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: alot.buffers
Traceback (most recent call last):
  File "/usr/lib64/python3.6/unittest/loader.py", line 462, in _find_test_path
    package = self._get_module_from_name(name)
  File "/usr/lib64/python3.6/unittest/loader.py", line 369, in _get_module_from_name
    __import__(name)
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/buffers/__init__.py", line 7, in <module>
    from .envelope import EnvelopeBuffer
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/buffers/envelope.py", line 9, in <module>
    from ..widgets.globals import HeadersList
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/widgets/globals.py", line 14, in <module>
    from ..db.attachment import Attachment
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/db/__init__.py", line 4, in <module>
    from .thread import Thread
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/db/thread.py", line 6, in <module>
    from .message import Message
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/db/message.py", line 10, in <module>
    from notmuch import NullPointerError
ModuleNotFoundError: No module named 'notmuch'


======================================================================
ERROR: alot.db (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: alot.db
Traceback (most recent call last):
  File "/usr/lib64/python3.6/unittest/loader.py", line 462, in _find_test_path
    package = self._get_module_from_name(name)
  File "/usr/lib64/python3.6/unittest/loader.py", line 369, in _get_module_from_name
    __import__(name)
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/db/__init__.py", line 4, in <module>
    from .thread import Thread
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/db/thread.py", line 6, in <module>
    from .message import Message
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/db/message.py", line 10, in <module>
    from notmuch import NullPointerError
ModuleNotFoundError: No module named 'notmuch'


======================================================================
ERROR: tests.commands.test_envelope (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: tests.commands.test_envelope
Traceback (most recent call last):
  File "/usr/lib64/python3.6/unittest/loader.py", line 428, in _find_test_path
    module = self._get_module_from_name(name)
  File "/usr/lib64/python3.6/unittest/loader.py", line 369, in _get_module_from_name
    __import__(name)
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/tests/commands/test_envelope.py", line 26, in <module>
    from alot.commands import envelope
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/commands/envelope.py", line 20, in <module>
    from . import globals
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/commands/globals.py", line 24, in <module>
    from .. import buffers
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/buffers/__init__.py", line 7, in <module>
    from .envelope import EnvelopeBuffer
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/buffers/envelope.py", line 9, in <module>
    from ..widgets.globals import HeadersList
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/widgets/globals.py", line 14, in <module>
    from ..db.attachment import Attachment
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/db/__init__.py", line 4, in <module>
    from .thread import Thread
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/db/thread.py", line 6, in <module>
    from .message import Message
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/db/message.py", line 10, in <module>
    from notmuch import NullPointerError
ModuleNotFoundError: No module named 'notmuch'


======================================================================
ERROR: tests.commands.test_global (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: tests.commands.test_global
Traceback (most recent call last):
  File "/usr/lib64/python3.6/unittest/loader.py", line 428, in _find_test_path
    module = self._get_module_from_name(name)
  File "/usr/lib64/python3.6/unittest/loader.py", line 369, in _get_module_from_name
    __import__(name)
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/tests/commands/test_global.py", line 24, in <module>
    from alot.commands import globals as g_commands
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/commands/globals.py", line 24, in <module>
    from .. import buffers
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/buffers/__init__.py", line 7, in <module>
    from .envelope import EnvelopeBuffer
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/buffers/envelope.py", line 9, in <module>
    from ..widgets.globals import HeadersList
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/widgets/globals.py", line 14, in <module>
    from ..db.attachment import Attachment
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/db/__init__.py", line 4, in <module>
    from .thread import Thread
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/db/thread.py", line 6, in <module>
    from .message import Message
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/db/message.py", line 10, in <module>
    from notmuch import NullPointerError
ModuleNotFoundError: No module named 'notmuch'


======================================================================
ERROR: tests.commands.test_init (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: tests.commands.test_init
Traceback (most recent call last):
  File "/usr/lib64/python3.6/unittest/loader.py", line 428, in _find_test_path
    module = self._get_module_from_name(name)
  File "/usr/lib64/python3.6/unittest/loader.py", line 369, in _get_module_from_name
    __import__(name)
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/tests/commands/test_init.py", line 10, in <module>
    from alot.commands import thread
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/commands/thread.py", line 20, in <module>
    from .globals import ExternalCommand
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/commands/globals.py", line 24, in <module>
    from .. import buffers
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/buffers/__init__.py", line 7, in <module>
    from .envelope import EnvelopeBuffer
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/buffers/envelope.py", line 9, in <module>
    from ..widgets.globals import HeadersList
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/widgets/globals.py", line 14, in <module>
    from ..db.attachment import Attachment
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/db/__init__.py", line 4, in <module>
    from .thread import Thread
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/db/thread.py", line 6, in <module>
    from .message import Message
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/db/message.py", line 10, in <module>
    from notmuch import NullPointerError
ModuleNotFoundError: No module named 'notmuch'


======================================================================
ERROR: tests.commands.test_thread (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: tests.commands.test_thread
Traceback (most recent call last):
  File "/usr/lib64/python3.6/unittest/loader.py", line 428, in _find_test_path
    module = self._get_module_from_name(name)
  File "/usr/lib64/python3.6/unittest/loader.py", line 369, in _get_module_from_name
    __import__(name)
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/tests/commands/test_thread.py", line 11, in <module>
    from alot.commands import thread
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/commands/thread.py", line 20, in <module>
    from .globals import ExternalCommand
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/commands/globals.py", line 24, in <module>
    from .. import buffers
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/buffers/__init__.py", line 7, in <module>
    from .envelope import EnvelopeBuffer
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/buffers/envelope.py", line 9, in <module>
    from ..widgets.globals import HeadersList
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/widgets/globals.py", line 14, in <module>
    from ..db.attachment import Attachment
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/db/__init__.py", line 4, in <module>
    from .thread import Thread
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/db/thread.py", line 6, in <module>
    from .message import Message
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/db/message.py", line 10, in <module>
    from notmuch import NullPointerError
ModuleNotFoundError: No module named 'notmuch'


======================================================================
ERROR: tests.db.test_envelope (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: tests.db.test_envelope
Traceback (most recent call last):
  File "/usr/lib64/python3.6/unittest/loader.py", line 428, in _find_test_path
    module = self._get_module_from_name(name)
  File "/usr/lib64/python3.6/unittest/loader.py", line 369, in _get_module_from_name
    __import__(name)
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/tests/db/test_envelope.py", line 25, in <module>
    from alot.db import envelope
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/db/__init__.py", line 4, in <module>
    from .thread import Thread
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/db/thread.py", line 6, in <module>
    from .message import Message
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/db/message.py", line 10, in <module>
    from notmuch import NullPointerError
ModuleNotFoundError: No module named 'notmuch'


======================================================================
ERROR: tests.db.test_manager (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: tests.db.test_manager
Traceback (most recent call last):
  File "/usr/lib64/python3.6/unittest/loader.py", line 428, in _find_test_path
    module = self._get_module_from_name(name)
  File "/usr/lib64/python3.6/unittest/loader.py", line 369, in _get_module_from_name
    __import__(name)
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/tests/db/test_manager.py", line 12, in <module>
    from alot.db.manager import DBManager
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/db/__init__.py", line 4, in <module>
    from .thread import Thread
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/db/thread.py", line 6, in <module>
    from .message import Message
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/db/message.py", line 10, in <module>
    from notmuch import NullPointerError
ModuleNotFoundError: No module named 'notmuch'


======================================================================
ERROR: tests.db.test_message (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: tests.db.test_message
Traceback (most recent call last):
  File "/usr/lib64/python3.6/unittest/loader.py", line 428, in _find_test_path
    module = self._get_module_from_name(name)
  File "/usr/lib64/python3.6/unittest/loader.py", line 369, in _get_module_from_name
    __import__(name)
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/tests/db/test_message.py", line 21, in <module>
    from alot.db import message
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/db/__init__.py", line 4, in <module>
    from .thread import Thread
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/db/thread.py", line 6, in <module>
    from .message import Message
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/db/message.py", line 10, in <module>
    from notmuch import NullPointerError
ModuleNotFoundError: No module named 'notmuch'


======================================================================
ERROR: tests.db.test_thread (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: tests.db.test_thread
Traceback (most recent call last):
  File "/usr/lib64/python3.6/unittest/loader.py", line 428, in _find_test_path
    module = self._get_module_from_name(name)
  File "/usr/lib64/python3.6/unittest/loader.py", line 369, in _get_module_from_name
    __import__(name)
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/tests/db/test_thread.py", line 22, in <module>
    from alot.db import thread
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/db/__init__.py", line 4, in <module>
    from .thread import Thread
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/db/thread.py", line 6, in <module>
    from .message import Message
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/db/message.py", line 10, in <module>
    from notmuch import NullPointerError
ModuleNotFoundError: No module named 'notmuch'


======================================================================
ERROR: tests.db.test_utils (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: tests.db.test_utils
Traceback (most recent call last):
  File "/usr/lib64/python3.6/unittest/loader.py", line 428, in _find_test_path
    module = self._get_module_from_name(name)
  File "/usr/lib64/python3.6/unittest/loader.py", line 369, in _get_module_from_name
    __import__(name)
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/tests/db/test_utils.py", line 26, in <module>
    from alot.db import utils
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/db/__init__.py", line 4, in <module>
    from .thread import Thread
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/db/thread.py", line 6, in <module>
    from .message import Message
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/db/message.py", line 10, in <module>
    from notmuch import NullPointerError
ModuleNotFoundError: No module named 'notmuch'


======================================================================
ERROR: tests.test_completion (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: tests.test_completion
Traceback (most recent call last):
  File "/usr/lib64/python3.6/unittest/loader.py", line 428, in _find_test_path
    module = self._get_module_from_name(name)
  File "/usr/lib64/python3.6/unittest/loader.py", line 369, in _get_module_from_name
    __import__(name)
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/tests/test_completion.py", line 10, in <module>
    from alot.completion.abooks import AbooksCompleter
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/completion/abooks.py", line 7, in <module>
    from ..db.utils import formataddr
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/db/__init__.py", line 4, in <module>
    from .thread import Thread
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/db/thread.py", line 6, in <module>
    from .message import Message
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/db/message.py", line 10, in <module>
    from notmuch import NullPointerError
ModuleNotFoundError: No module named 'notmuch'


======================================================================
ERROR: tests.widgets.test_globals (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: tests.widgets.test_globals
Traceback (most recent call last):
  File "/usr/lib64/python3.6/unittest/loader.py", line 428, in _find_test_path
    module = self._get_module_from_name(name)
  File "/usr/lib64/python3.6/unittest/loader.py", line 369, in _get_module_from_name
    __import__(name)
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/tests/widgets/test_globals.py", line 22, in <module>
    from alot.widgets import globals as globals_
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/widgets/globals.py", line 14, in <module>
    from ..db.attachment import Attachment
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/db/__init__.py", line 4, in <module>
    from .thread import Thread
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/db/thread.py", line 6, in <module>
    from .message import Message
  File "/var/tmp/portage/mail-client/alot-0.9.1/work/alot-0.9.1/alot/db/message.py", line 10, in <module>
    from notmuch import NullPointerError
ModuleNotFoundError: No module named 'notmuch'


----------------------------------------------------------------------
Ran 190 tests in 0.631s
```

So you'll have to add it here. Most likely to `RDEPEND` too.

File: mail-client/alot/alot-0.9.1.ebuild [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+RDEPEND="
+	app-crypt/gpgme[python,${PYTHON_USEDEP}]
+	dev-python/configobj[${PYTHON_USEDEP}]
+	dev-python/python-magic[${PYTHON_USEDEP}]
+	dev-python/urwid[${PYTHON_USEDEP}]
+	dev-python/urwidtrees[${PYTHON_USEDEP}]
+	>=dev-python/twisted-18.4[${PYTHON_USEDEP}]
+	net-mail/mailbase
+	net-mail/notmuch[crypt,python]
```
Comment (juippis) [inline mail-client/alot/alot-0.9.1.ebuild:28]: I still think you'll need to add `PYTHON_USEDEP` here, the test phase clearly implies that. However with notmuch-30 I'm not sure how it affects the runtime, you probably have a better idea? How is notmuch used by alot? It should be broken with 30 currently. Maybe the version can be restricted for now, or we wait until notmuch-30 has been fixed?

Comment (GuillaumeSeren) [reply mail-client/alot/alot-0.9.1.ebuild:28]: > I still think you'll need to add PYTHON_USEDEP here, the test phase clearly implies that. However with notmuch-30 I'm not sure how it affects the runtime, you probably have a better idea?

I am not really sure, what I can say alot use the python binding provided by notmuch,
I used to think that when notmuch is installed it is build with all python version, maybe I am wrong.

> Maybe the version can be restricted for now, or we wait until notmuch-30 has been fixed?
Yes I contacted upstream (https://github.com/pazz/alot/issues/1539) it seems,
a PR is already going on to port to 'notmuch2' module (https://github.com/pazz/alot/pull/1511).

So with all that, I think I probably fix to '<=net-mail/notmuch-0.30[crypt,python,${PYTHON_USEDEP}]',
until everything is fixed.

**Comment by GuillaumeSeren** [comment]
Hey,
I have fixed my chroot test setup, and I detected one new issue (urwid).

I tested it against a stage3 image, and it works building with tests and doc enabled.

**Comment by janko-j1** [comment]
> Hey,
> I have fixed my chroot test setup, and I detected one new issue (urwid).
> 
> I tested it against a stage3 image, and it works building with tests and doc enabled.

Did you try alot without doc?-)

**Comment by GuillaumeSeren** [comment]
> Did you try alot without doc?-)

:octocat: 
I tested with doc only without test, but I am testing only test right now.

**Comment by GuillaumeSeren** [comment]
You were right @janko-j1
I fixed the sphinx dep into a BDEPEND.

File: dev-python/urwid/urwid-2.1.0.ebuild [hunk 13]
```diff
@@ -17,6 +17,10 @@ SLOT="0"
 KEYWORDS="amd64 ~arm ~arm64 ~ia64 ~mips ppc ppc64 ~sparc x86 ~amd64-linux ~x86-linux"
 IUSE="examples"
 
+PATCHES=(
+	"${FILESDIR}/2.1-Ignore-resetting-to-invalid-locale-Closes-377.patch"
```
Comment (juippis) [inline dev-python/urwid/urwid-2.1.0.ebuild:21]: Don't touch the stable ebuild here at all, I'm not comfortable with it.

File: dev-python/urwid/urwid-2.1.1.ebuild [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+HOMEPAGE="http://urwid.org/ https://pypi.org/project/urwid/"
+SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
+
+LICENSE="LGPL-2.1"
+SLOT="0"
+KEYWORDS="amd64 ~arm ~arm64 ~ia64 ~mips ppc ppc64 ~sparc x86 ~amd64-linux ~x86-linux"
+IUSE="examples"
+
+PATCHES=(
+	"${FILESDIR}/2.1-Ignore-resetting-to-invalid-locale-Closes-377.patch"
```
Comment (juippis) [inline dev-python/urwid/urwid-2.1.1.ebuild:21]: Rename the patch file better, like `urwid-2.1.1-ignore-resetting-to-invalid-locale.patch`.

File: dev-python/urwid/urwid-2.1.1.ebuild [hunk 15]
```diff
@@ -0,0 +1,46 @@
...
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{6..9} )
+PYTHON_REQ_USE="ncurses"
+
+inherit distutils-r1
+
+DESCRIPTION="Curses-based user interface library for Python"
+HOMEPAGE="http://urwid.org/ https://pypi.org/project/urwid/"
```
Comment (juippis) [inline dev-python/urwid/urwid-2.1.1.ebuild:12]: Please also add the Github page here, they seem to track issues there.

File: dev-python/urwid/urwid-2.1.1.ebuild [hunk 16]
```diff
@@ -0,0 +1,46 @@
...
+SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
+
+LICENSE="LGPL-2.1"
+SLOT="0"
+KEYWORDS="amd64 ~arm ~arm64 ~ia64 ~mips ppc ppc64 ~sparc x86 ~amd64-linux ~x86-linux"
```
Comment (juippis) [inline dev-python/urwid/urwid-2.1.1.ebuild:17]: Don't commit straight to stable.

File: dev-python/urwid/urwid-2.1.0.ebuild (hunk 13 above)
Comment (GuillaumeSeren) [reply dev-python/urwid/urwid-2.1.0.ebuild:21]: Ok I am creating a new urwid-2.1.0-r1 with the fix in it

File: dev-python/urwid/urwid-2.1.1.ebuild (hunk 14 above)
Comment (GuillaumeSeren) [reply dev-python/urwid/urwid-2.1.1.ebuild:21]: Ok

Comment (GuillaumeSeren) [reply dev-python/urwid/urwid-2.1.1.ebuild:21]: done

File: dev-python/urwid/urwid-2.1.1.ebuild (hunk 15 above)
Comment (GuillaumeSeren) [reply dev-python/urwid/urwid-2.1.1.ebuild:12]: ok

File: dev-python/urwid/urwid-2.1.1.ebuild (hunk 16 above)
Comment (GuillaumeSeren) [reply dev-python/urwid/urwid-2.1.1.ebuild:17]: Ok

**Comment by GuillaumeSeren** [comment]
> And building urwid with `[doc]` doesn't work, although it doesn't work for current tree version either, could you look to fix it?
I did open a issue on that on gentoo and reported upstream.
https://bugs.gentoo.org/736410

> Please make a new PR for urwid if you absolutely need it, but to me looks like alot can work with 2.1.
It also break alot on master now
https://github.com/gentoo/gentoo/pull/17059

**Comment by GuillaumeSeren** [comment]
Thank you @juippis for the feedback and merge !
